platforms/jvm/scala/src/main/java/org/gradle/api/tasks/ScalaRuntime.java

```diff
@@ -60,12 +61,14 @@
  */
 public abstract class ScalaRuntime {
 
-    private final Project project;
+    private final ProjectInternal project;
+    private final DependencyFactory dependencyFactory;
     private final JvmPluginServices jvmPluginServices;
 
     public ScalaRuntime(Project project) {
-        this.project = project;
-        this.jvmPluginServices = ((ProjectInternal) project).getServices().get(JvmPluginServices.class);
+        this.project = (ProjectInternal) project;
+        this.dependencyFactory = this.project.getServices().get(DependencyFactory.class);
+        this.jvmPluginServices = this.project.getServices().get(JvmPluginServices.class);
```

Comment:
I think you can add injectable services here, like:

```
@Inject 
protected abstract JvmPluginServices getJvmPluginServices();
```

Reply:
This code was since removed